akka-docs/rst/java/code/jdocs/akka/typed/IntroSpec.java

```diff
@@ -0,0 +1,229 @@
+          sessions.forEach(s -> s.tell(mp));
+          return Actor.same();
+        } else {
+          return Actor.unhandled();
+        }
+      });
+    }
+    //#chatroom-behavior
+
+    public static Behavior<Command> mutableChatRoom() {
+      return Actor.deferred(c -> {
+        final List<ActorRef<SessionEvent>> sessions = new ArrayList<ActorRef<SessionEvent>>();
+        int postCount = 0;
+
+        return Actor.stateful((ctx, msg) -> {
+          if (msg instanceof GetSession) {
+            GetSession getSession = (GetSession) msg;
+            ActorRef<PostMessage> wrapper = ctx.createAdapter(p ->
+              new PostSessionMessage(getSession.screenName, p.message));
+            getSession.replyTo.tell(new SessionGranted(wrapper));
+            sessions.add(getSession.replyTo);
+            return Actor.same();
+          } else if (msg instanceof PostSessionMessage) {
+            PostSessionMessage post = (PostSessionMessage) msg;
+            MessagePosted mp = new MessagePosted(post.screenName, post.message);
+            // Local variable postCount defined in an enclosing scope must be final or effectively final
+            //postCount += 1;
+            sessions.forEach(s -> s.tell(mp));
+            return Actor.same();
+          } else {
+            return Actor.unhandled();
+          }
+        });
+      });
+    }
+
+    public static Behavior<Command> mutableChatRoom2() {
+      // TODO Type safety: The expression of type IntroSpec.ChatRoom.MutableChatRoom2 needs unchecked conversion to
+      //      conform to Behavior<IntroSpec.ChatRoom.Command>
+      return Actor.deferred(ctx -> new MutableChatRoom2());
+    }
+
+    public static class MutableChatRoom2 extends ExtensibleBehavior<Command> {
+      final List<ActorRef<SessionEvent>> sessions = new ArrayList<ActorRef<SessionEvent>>();
+      int postCount = 0;
+
+      @Override
+      public Behavior<Command> message(akka.typed.ActorContext<Command> ctx, Command msg) {
+        if (msg instanceof GetSession) {
+          GetSession getSession = (GetSession) msg;
+          ActorRef<PostMessage> wrapper = ctx.createAdapter(p ->
+            new PostSessionMessage(getSession.screenName, p.message));
+          getSession.replyTo.tell(new SessionGranted(wrapper));
+          sessions.add(getSession.replyTo);
+          return Actor.same();
+        } else if (msg instanceof PostSessionMessage) {
+          PostSessionMessage post = (PostSessionMessage) msg;
+          MessagePosted mp = new MessagePosted(post.screenName, post.message);
+          postCount += 1;
+          sessions.forEach(s -> s.tell(mp));
+          return Actor.same();
+        } else {
+          return Actor.unhandled();
+        }
+      }
+
+      @Override
+      public Behavior<Command> management(akka.typed.ActorContext<Command> ctx, Signal msg) {
+        return Actor.same();
+      }
+    }
+
+    public static Behavior<Command> mutableChatRoom3() {
```

Comment:
Below is my API suggestion.
Yeah, it has many similarities to current untyped Actor API, and I think that is another advantage. Migration will be easier.

I have used Java to illustrate this, but same thing makes sense in Scala.